public/app/features/explore/ExploreGraphNGPanel.tsx

```diff
@@ -0,0 +1,165 @@
+import {
+  AbsoluteTimeRange,
+  applyFieldOverrides,
+  createFieldConfigRegistry,
+  DataFrame,
+  dateTime,
+  Field,
+  FieldColorModeId,
+  FieldConfigSource,
+  GrafanaTheme,
+  TimeZone,
+} from '@grafana/data';
+import {
+  Collapse,
+  DrawStyle,
+  GraphNG,
+  GraphNGLegendEvent,
+  Icon,
+  LegendDisplayMode,
+  TooltipPlugin,
+  useStyles,
+  useTheme,
+  ZoomPlugin,
+} from '@grafana/ui';
+import { defaultGraphConfig, getGraphFieldConfig } from 'app/plugins/panel/timeseries/config';
+import { hideSeriesConfigFactory } from 'app/plugins/panel/timeseries/hideSeriesConfigFactory';
+import { ContextMenuPlugin } from 'app/plugins/panel/timeseries/plugins/ContextMenuPlugin';
+import { ExemplarsPlugin } from 'app/plugins/panel/timeseries/plugins/ExemplarsPlugin';
+import { css, cx } from 'emotion';
+import React, { useCallback, useMemo, useState } from 'react';
+import { splitOpen } from './state/main';
+import { getFieldLinksForExplore } from './utils/links';
+
+const MAX_NUMBER_OF_TIME_SERIES = 20;
+
+interface Props {
+  data: DataFrame[];
+  annotations?: DataFrame[];
+  isLoading: boolean;
+  width: number;
+  absoluteRange: AbsoluteTimeRange;
+  timeZone: TimeZone;
+  onUpdateTimeRange: (absoluteRange: AbsoluteTimeRange) => void;
+  splitOpenFn: typeof splitOpen;
+}
+
+export function ExploreGraphNGPanel({
+  width,
+  data,
+  timeZone,
+  absoluteRange,
+  onUpdateTimeRange,
+  isLoading,
+  annotations,
+  splitOpenFn,
+}: Props) {
+  const theme = useTheme();
+  const [showAllTimeSeries, setShowAllTimeSeries] = useState(false);
+  const [fieldConfig, setFieldConfig] = useState<FieldConfigSource>({
```

Comment:
Me and @zoltanbedi talked back and forth about moving this to redux and apply the fieldOverrides in the runQuery pipeline via decorators. What would be the benefit of doing that compared to keep it like this @torkelo @dprokop.

One benefit that I can think of is that we would be able to use the fieldOverrides on other parts of Explore. But please jump in here with some more feedback since I don't have good knowledge about the Explore code.

Reply:
> Me and @zoltanbedi talked back and forth about moving this to redux and apply the fieldOverrides in the runQuery pipeline via decorators. What would be the benefit of doing that compared to keep it like this 

As it's already done there for table (sort of via setting display processor). 

And doing there would be it is only done on data updates, not on other rendering triggers. But if its ieasier to use local graph component state & memoize the applyfield overrides call in the explore graph component then I am fine doing it inside the component if that is quicker/easier/faster now

Reply:
@zoltanbedi I guess it is up to you :)

Reply:
We thought that we leave it like this for now and will refactor it later if needed. Thanks for the input anyway!